crypto/md4/md4_dgst.c

```diff
@@ -22,6 +30,10 @@
 
 int MD4_Init(MD4_CTX *c)
 {
+    if (FIPS_mode()) {
+        FIPSerr(ERR_LIB_FIPS, FIPS_R_NON_FIPS_METHOD);
```

Comment:
For MD4 and MD5, I don't see the corresponding pieces in Update and Final that we have for MD2 here - at least not in this PR's context. Are we missing them, or did we already have them before this PR?

Reply:
The Update, Transform and Final methods are constructed via inline functions in crypto/md32_common.h.

We update the HASH_UPDATE, HASH_TRANSFORM and HASH_FINAL definitions in  crypto/md32_common.h to add the needed code.

Reply:
git show ea3f8d0c1556f908f8356b28c7dd22b3f300adcc "Prevent use of SHA1, MD2, MD4, MD5, ripemd160, sm3 in libcrypto when in FIPS mode." is the commit that does this.